core/pipeline/label.go

```diff
@@ -81,10 +82,20 @@ func LoadClientLabels(engineVersion string) []Label {
 		},
 	}
 
+	machineID, err := machineid.ProtectedID("dagger")
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to resolve machine id: %v\n", err)
```

Comment:
None of the files the package looks for are present on my Linux machine, so this will always log, and probably mess up the TUI. Guessing this is the case for anyone using Nix who hasn't gone out of their way to set a machine ID. Dunno about Ubuntu/etc.

Maybe this should just be best-effort, without even logging if it fails to collect a machine ID.

https://github.com/denisbrodbeck/machineid/blob/f07179bb15a0bec252453468aa607c670f5c12b5/id_linux.go#L6-L11

Reply:
Oh wait, it does check for `/etc/machine-id` which exists, for some reason I checked `/etc/hostid`.

Still a bit worried about the `os.Stderr` log, since it'll disrupt the TUI. Could use `progrock.FromContext(ctx).Warn` instead if you want it to show up there, or `Debug` to show it with `--debug`.